com.unity.shadergraph/Editor/Data/Nodes/Channel/SwizzleNode.cs

```diff
@@ -27,146 +26,174 @@ public override bool hasPreview
+            bool MaskInputIsValid = true;
+            char[] MaskChars = _maskInput.ToCharArray();
+            char[] AllChars = { 'x', 'y', 'z', 'w', 'r', 'g', 'b', 'a' };
+            List<char> CurrentChars = new List<char>();
+            for (int i = 0; i < InputValueSize; i++)
+            {
+                CurrentChars.Add(AllChars[i]);
+                CurrentChars.Add(AllChars[i + 4]);
             }
-        }
-
-        [SerializeField]
-        TextureChannel m_BlueChannel;
 
-        [ChannelEnumControl("Blue Out")]
-        public TextureChannel blueChannel
-        {
-            get { return m_BlueChannel; }
-            set
+            foreach (char c in MaskChars)
             {
-                if (m_BlueChannel == value)
-                    return;
-
-                m_BlueChannel = value;
-                Dirty(ModificationScope.Node);
+                if (!CurrentChars.Contains(c))
+                {
+                    MaskInputIsValid = false;
+                }
             }
-        }
-
-        [SerializeField]
-        TextureChannel m_AlphaChannel;
-
-        [ChannelEnumControl("Alpha Out")]
-        public TextureChannel alphaChannel
-        {
-            get { return m_AlphaChannel; }
-            set
+            if (MaskChars.Length <= 0 || MaskChars.Length > 4)
             {
-                if (m_AlphaChannel == value)
-                    return;
-
-                m_AlphaChannel = value;
-                Dirty(ModificationScope.Node);
+                MaskInputIsValid = false;
             }
+            //Convert "rgba" input to "xyzw" to avoid mismathcing
+            if (MaskInputIsValid)
+            {
+                char[] rgba = { 'r', 'g', 'b', 'a' };
+
+                for (int i = 0; i < MaskChars.Length; i++)
+                {
+                    if (rgba.Contains(MaskChars[i]))
+                    {
+                        MaskChars[i] = mask_map[MaskChars[i]];
+                    }
+                }
+                convertedMask = new string(MaskChars);
+            }
+            return MaskInputIsValid;
         }
 
-        void ValidateChannelCount()
+        public sealed override void UpdateNodeAfterDeserialization()
         {
-            var channelCount = SlotValueHelper.GetChannelCount(FindInputSlot<MaterialSlot>(InputSlotId).concreteValueType);
-            if ((int)redChannel >= channelCount)
-                redChannel = TextureChannel.Red;
-            if ((int)greenChannel >= channelCount)
-                greenChannel = TextureChannel.Red;
-            if ((int)blueChannel >= channelCount)
-                blueChannel = TextureChannel.Red;
-            if ((int)alphaChannel >= channelCount)
-                alphaChannel = TextureChannel.Red;
+            AddSlot(new DynamicVectorMaterialSlot(InputSlotId, kInputSlotName, kInputSlotName, SlotType.Input, Vector4.zero));
+            switch (_maskInput.Length)
+            {
+                default:
```

Comment:
This switch statement looks backwards to me, and it makes me stop and wonder if there's a reason you did it that way (usually numbers go in ascending order and default comes last). Is there a reason?